vp8_remb_simulcast.c

```diff
@@ -0,0 +1,571 @@
+				nextLower--;
+			}
+		}
+
+		if(nextLower < 0)
+			nextLower = 0;
+	}
+
+	if(nextHigher < 5) {
+		/* Increase the ramp Position */
+		nextHigher++;
+
+		/* Check that a preselected temporal from the client does not conflict our ramping up */
+		if(nextHigher % 2) {
+			/* 1,3,5 are the layers where we switch to 30fps, */
+			if (pSim->templayer_target == 0 || pSim->templayer_target == 1) {
+				/* if the client is requesting a lower temporal (7fps, 15fps)
+				 * we cannot use it and need to switch directly to the next substream */
+				nextHigher++;
+			}
+		}
+
+		/* whats the currently selected substream target layer from the client? (-1 means not defined, so we say highest) */
+		int substream_target = pSim->substream_target;
+		if (substream_target == -1)
+			substream_target = 2;
+
+		/* Check that a preselected substream layer from the client does not conflict our ramping up
+		 * Ramping up next would be substream high (2), but the client requested low or mid */
+		if (nextHigher > 3 && substream_target < 2)
+			nextHigher = 0;
+
+		/* Ramping up next would be substream mid (1), but the client requested low */
+		else if (nextHigher > 1 && substream_target < 1)
+			nextHigher = 0;
+	}
+
+	*pNextLower = (uint32_t)nextLower;
+	*pNextHigher = (uint32_t)nextHigher;
+}
+
+/*
+ * Sets the flags (limits) for a given ramp position in the simulcasting context
+ *
+ * @param pRembContext - the currently used REMB simulcasting context
+ * @param rampPos - The ramp position to set the flags for
+ * @returns - true in case the value was set or false
+ */
+gboolean janus_vp8_remb_simulcast_set_flags_for_ramp_position(janus_rtp_simulcasting_remb_context *pRembContext, uint32_t position) {
+	switch(position) {
+		case 0: /* Substream 0 - Temporal 15fps */
+			pRembContext->substream_limit_by_remb = 0;
+			pRembContext->templayer_limit_by_remb = 1;
+			return TRUE;
+		case 1: /* Substream 0 - Temporal 30fps */
+			pRembContext->substream_limit_by_remb = 0;
+			pRembContext->templayer_limit_by_remb = -1;
+			return TRUE;
+		case 2: /* Substream 1 - Temporal 15fps */
+			pRembContext->substream_limit_by_remb = 1;
+			pRembContext->templayer_limit_by_remb = 1;
+			return TRUE;
+		case 3: /* Substream 1 - Temporal 30fps */
+			pRembContext->substream_limit_by_remb = 1;
+			pRembContext->templayer_limit_by_remb = -1;
+			return TRUE;
+		case 4: /* Substream 2 - Temporal 15fps */
+			pRembContext->substream_limit_by_remb = -1;
+			pRembContext->templayer_limit_by_remb = 1;
+			return TRUE;
+		case 5: /* Substream 2 - Temporal 30fps */
+			pRembContext->substream_limit_by_remb = -1;
+			pRembContext->templayer_limit_by_remb = -1;
+			return TRUE;
+		default:
+			return FALSE;
+	}
+}
+
+/*
+ * Retrieve diagnostic log text for a substream layer (L,M,H)
+ *
+ * @param substream - the substream layer we want to have the text for
+ * @returns - L,M,H for the different substreams or unknown if the value could not get converted
```

Comment:
To match the style we use in the core (e.g., in Admin API), these should be "l", "m" and "h", not the capitalized versions.